app/src/main/java/de/westnordost/streetcomplete/overlays/surface/PathSurfaceOverlay.kt

```diff
@@ -0,0 +1,211 @@
+package de.westnordost.streetcomplete.overlays.surface
+
+import de.westnordost.streetcomplete.R
+import de.westnordost.streetcomplete.data.osm.mapdata.Element
+import de.westnordost.streetcomplete.data.osm.mapdata.MapDataWithGeometry
+import de.westnordost.streetcomplete.data.osm.mapdata.filter
+import de.westnordost.streetcomplete.osm.ALL_PATHS
+import de.westnordost.streetcomplete.osm.ALL_ROADS
+import de.westnordost.streetcomplete.osm.isPrivateOnFoot
+import de.westnordost.streetcomplete.osm.sidewalk.Sidewalk
+import de.westnordost.streetcomplete.osm.sidewalk.createSidewalkSides
+import de.westnordost.streetcomplete.osm.surface.CyclewayFootwaySurfaces
+import de.westnordost.streetcomplete.osm.surface.CyclewayFootwaySurfacesWithNote
+import de.westnordost.streetcomplete.osm.surface.SingleSurface
+import de.westnordost.streetcomplete.osm.surface.SingleSurfaceWithNote
+import de.westnordost.streetcomplete.osm.surface.Surface
+import de.westnordost.streetcomplete.osm.surface.Surface.PAVED_AREA
+import de.westnordost.streetcomplete.osm.surface.Surface.PAVED_ROAD
+import de.westnordost.streetcomplete.osm.surface.Surface.UNPAVED_AREA
+import de.westnordost.streetcomplete.osm.surface.Surface.UNPAVED_ROAD
+import de.westnordost.streetcomplete.osm.surface.SurfaceMissing
+import de.westnordost.streetcomplete.osm.surface.SurfaceMissingWithNote
+import de.westnordost.streetcomplete.osm.surface.UNDERSPICIFED_SURFACES
+import de.westnordost.streetcomplete.osm.surface.associatedKeysToBeRemovedOnChange
+import de.westnordost.streetcomplete.osm.surface.createSurfaceStatus
+import de.westnordost.streetcomplete.overlays.Color
+import de.westnordost.streetcomplete.overlays.Overlay
+import de.westnordost.streetcomplete.overlays.PolygonStyle
+import de.westnordost.streetcomplete.overlays.PolylineStyle
+import de.westnordost.streetcomplete.overlays.StrokeStyle
+import de.westnordost.streetcomplete.overlays.Style
+import de.westnordost.streetcomplete.quests.surface.AddPathSurface
+import de.westnordost.streetcomplete.quests.surface.GENERIC_AREA_SURFACES
+import de.westnordost.streetcomplete.quests.surface.GENERIC_ROAD_SURFACES
+
+class PathSurfaceOverlay : Overlay {
```

Comment:
General comment regarding paths + sidewalks:

Do you handle paths with sidewalks? E.g. `highway=cycleway` + `sidewalk=right`?
Can you think of a way to support this without complicating the UI (too much)?

I do support it now for sidewalks in the WIP cycleway-overlay branch.

Reply:
> Do you handle paths with sidewalks? E.g. highway=cycleway + sidewalk=right?

no

>  Can you think of a way to support this without complicating the UI (too much)?

not really and backed will be even worse. `surface:note` `footway:surface:note` `cycleway:surface:note` are providing already interesting complexity

> General remark: +2700 lines of code added sounds a bit much, even with all those tests added. But let's see .... (after reading it:) well, it is also huge. I didn't spot any things that could obviously be cut down by a lot.

yeaah

>  Can you think of a way to support this without complicating the UI (too much)?

Maybe treat them as roads? (main surface editable in road overlay, sidewalks editable in path overlay)

Reply:
added to test, added to PR notes